trace-mini-agent/src/config.rs

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use std::env;
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    pub api_key: String,
+    pub gcp_function_name: Option<String>,
```

Comment:
I don't think you need an optional here?

Reply:
I kept it Option<> for when the mini agent + config is used for non-serverless usecases where there is no gcp_function_name